### PR TITLE
Custom wake anim

### PR DESCRIPTION
**Pull request by Super-StarX**
custom the wake anim when unit on water

### CODE REVIEW

**Review by Metadorius — COMMENTED** [review]
I assume the PR is in progress, it would be better to mark it as draft. Left some recommendations.

File: src/Ext/TechnoType/Body.cpp [hunk 1]
```diff
@@ -115,6 +115,16 @@ TechnoTypeClass* TechnoTypeExt::GetTechnoType(ObjectTypeClass* pType)
 	return nullptr;
 }
 
+AnimTypeClass* TechnoTypeExt::GetWakeAnim(LocomotionClass* pLoco)
```
[the remark this answers is not in the document]
Comment (Super-StarX) [reply src/Ext/TechnoType/Body.cpp:118]: but I cant find LinkedTo in it

[the remark this answers is not in the document]
Comment (Metadorius) [reply src/Ext/TechnoType/Body.cpp:118]: Hmm you're right, nevermind then. I was under impression some getter for LinkedTo exists for `ILocomotion`.

File: src/Ext/TechnoType/Hooks.cpp [hunk 2]
```diff
@@ -364,3 +364,22 @@ DEFINE_HOOK(0x4DB157, FootClass_DrawVoxelShadow_TurretShadow, 0x8)
...
+	GET(ILocomotion*, pILoco, ESI);
+
+	if (auto pLoco = static_cast<LocomotionClass*>(pILoco))
+	{
+		if (auto pAnim = TechnoTypeExt::GetWakeAnim(pLoco))
+		{
+			R->EDX(pAnim);
+			return R->Origin() + 0xC;
+		}
+	}
```
Comment (MortonPL) [inline src/Ext/TechnoType/Hooks.cpp:381]:
```suggestion
    const auto pLoco = static_cast<LocomotionClass*>(pILoco);
    if (!pLoco)
        return 0;
    
    const auto pAnim = TechnoTypeExt::GetWakeAnim(pLoco);
    if (!pAnim)
        return 0;

	R->EDX(pAnim);
	return R->Origin() + 0xC;
```

Avoid nesting blocks if possible. Use `const` when possible.

File: src/Ext/TechnoType/Body.cpp [hunk 3]
```diff
@@ -115,6 +115,16 @@ TechnoTypeClass* TechnoTypeExt::GetTechnoType(ObjectTypeClass* pType)
...
+{
+	if (auto pThis = pLoco->LinkedTo)
+		if (auto pExt = TechnoTypeExt::ExtMap.Find(pThis->GetTechnoType()))
+			if (auto pAnim = pExt->Wake)
+				return pAnim;
```
Comment (MortonPL) [inline src/Ext/TechnoType/Body.cpp:123]: Nested braceless blocks are not allowed within braceless blocks. Avoid nesting blocks if possible. Use `const` when possible.

**Review by MortonPL — CHANGES_REQUESTED** [review]
Found two style issues. Please add credits, documentation and "What's New" for this feature.

**Comment by Metadorius** [comment]
Closing in favor of #1258 containing an improved version.

**Pull request closed without merging**